### PR TITLE
feat(retail): generate library

### LINKED ISSUE #7971: Retail API: `retail.googleapis.com`

### PR DESCRIPTION
**Pull request by scotthart**
fixes  #7971 

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7972)

### CODE REVIEW

File: google/cloud/retail/internal/catalog_stub_factory.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// limitations under the License.
+
+// Generated by the Codegen C++ plugin.
+// If you make any local changes, they will be lost.
+// source: google/cloud/retail/v2/catalog_service.proto
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_RETAIL_INTERNAL_CATALOG_STUB_FACTORY_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_RETAIL_INTERNAL_CATALOG_STUB_FACTORY_H
+
+#include "google/cloud/retail/catalog_connection.h"
```
Comment (coryan) [inline google/cloud/retail/internal/catalog_stub_factory.h:22]: I just merged a change that removes this `#include`.  You will need to rebase, run `ci/cloudbuild/build.sh -t generate-libraries-pr` and merge any changes.

File: external/googleapis/CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,7 @@ set(EXTERNAL_GOOGLEAPIS_SOURCE
 
 set(EXTERNAL_GOOGLEAPIS_PROTO_FILES
     "google/api/http.proto"
+    "google/api/httpbody.proto"
```
Comment (coryan) [inline external/googleapis/CMakeLists.txt:28]: I would have separated this to a different PR, but no worries.

**Review by coryan — CHANGES_REQUESTED** [review]
Unfortunately you will need to rebase and regenerate.  Even if the builds pass, the post-merge builds will probably break.

**Review by scotthart — COMMENTED** [review]
Rebased and regenerated

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7972#-:-MtnCQSu5Zg-Gs7N6N65:b7epm5q)* status: 0 of 148 files reviewed, 1 unresolved discussion (waiting on @coryan and @scotthart)

**Review by coryan — CHANGES_REQUESTED** [review]
> Reviewed 123 of 148 files at r1, 25 of 25 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7972)* status: all files reviewed, 2 unresolved discussions (waiting on @scotthart)

___
*[external/googleapis/CMakeLists.txt, line 529 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7972#-MtnGY5rEM7cfqAXTTCu:-MtnGY5s92C80oTXfRL3:blkbekb) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/193fb8b48610b31ead4875dfa8eb07296ce89b16/external/googleapis/CMakeLists.txt#L529)):*
> ```CMake
>            " google_cloud_cpp_api_field_behavior_protos"
>            " google_cloud_cpp_api_http_protos"
>            " google_cloud_cpp_api_httpbody_protos"
> ```

This list is for backwards compatibility, we do not need to add new libraries to it.  Please remove.
___
*[google/cloud/retail/quickstart/README.md, line 160 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7972#-MtnHune2dRxs90dHtpV:-MtnHune2dRxs90dHtpW:b-6gip5v) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/193fb8b48610b31ead4875dfa8eb07296ce89b16/google/cloud/retail/quickstart/README.md#L160)):*
> ```Markdown
> 
> [bazel-install]: https://docs.bazel.build/versions/main/install.html
> [quickstart-link]: https://cloud.google.com/retail/docs/quickstart
> ```

This link is broken, maybe: https://cloud.google.com/retail/docs/setting-up ??



_Code quote:_
```Markdown
 https://cloud.google.com/retail/docs/quickstart
```

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7972)* status: 146 of 148 files reviewed, 2 unresolved discussions (waiting on @coryan)

___
*[external/googleapis/CMakeLists.txt, line 529 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7972#-MtnGY5rEM7cfqAXTTCu:-MtnT1HO9FbM2cwHl3cz:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/193fb8b48610b31ead4875dfa8eb07296ce89b16/external/googleapis/CMakeLists.txt#L529)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

This list is for backwards compatibility, we do not need to add new libraries to it.  Please remove.
</blockquote></details>

Done.
___
*[google/cloud/retail/quickstart/README.md, line 160 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7972#-MtnHune2dRxs90dHtpV:-MtnT1tzBbC4V--JQBiw:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/193fb8b48610b31ead4875dfa8eb07296ce89b16/google/cloud/retail/quickstart/README.md#L160)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

This link is broken, maybe: https://cloud.google.com/retail/docs/setting-up ??
</blockquote></details>

Done.

**Review by coryan — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7972#-:-MtoUVt07_ymkItZYScI:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @scotthart)

**Pull request merged**